examples/hackernews/front-end/src/App.tsx

```diff
@@ -149,13 +155,19 @@ function Post() {
     }
   }
 
-  useEffect(() => {
-    const intervalId = setInterval(() => {
-      getPost(post_id).then(setPost);
-    }, 1000);
-
-    return () => clearInterval(intervalId);
-  }, []);
+  useEffect(
+    () => {
+      const intervalId = setInterval(() => {
+        getPost(post_id).then(setPost);
+      }, 1000);
+
+      return () => clearInterval(intervalId);
+    },
+    // Specify no dependencies to react since reevaluation of the callback
+    // is triggered after a timeout by setInterval.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [],
```

Comment:
Here, the linter wants
```suggestion
    [post_id],
```

Reply:
For this case, my understanding of react (which could be wrong!) is that adding the `post_id` dep here would have the effect of triggering reevaluation sooner than the timeout in case `post_id` changes. But, and maybe I'm confused, isn't `post_id` constant, and so it makes no difference? Either way, it seems ok to me to add the dep here and appease the linter.

Reply:
That's my understanding of react/useEffect semantics here as well.  But, `post_id` comes from `useParams`, which is React's way of accessing URL parameters and thus not necessarily constant -- so I think it's right to specify it as a dependency here. 

The code looks similar enough to [this example](https://react.dev/reference/react/useEffect#updating-state-based-on-previous-state-from-an-effect) from the react docs that I suspect that the empty deps array is a relic of some earlier version that was set up that same way.

Reply:
Ok thanks, I will change this. I am fundamentally confused by react. What I expect is that at the point that `useEffect` is called, the function argument gets evaluated to a closure that binds `post_id`, which has type `number` and is already evaluated. So I don't see how react can treat it as anything other than a constant number value.

Reply:
Within a single render, yes, `post_id` is a constant number.  But if the params change, that will trigger a re-render with some new `post_id`!